strapi/strapi
# Add the values object to the Array utils

### What does it do?

- Add the base entity service interface
- [**experiment/internal use only**]: Allow plugins to extend the entity service params to imitate the **internal** ES decorator behavior.

### Why is it needed?

- Necessary to start adding ES methods type definitions 

### How to test it?

- `cd examples/getstarted`
- `yarn strapi ts:generate-types`
- Create `src/index.d.ts`
- Add the following
	```ts
	import type { EntityService } from '@strapi/strapi';
	
	declare module '@strapi/strapi' {
	  export module Shared {
	    export interface PluginActivation {
	      i18n: { localized: true };
	    }
	
	    export interface EntityServicePluginReadParams {
	      i18n: { locale?: string };
	    }
	
	    export interface EntityServicePluginWriteParams {
	      i18n: { locale?: string };
	    }
	  }
	}
	
	const userParams: EntityService.Params.ReadWrite<'admin::user'> = {
	  /* locale shouldn't be suggested here */
	};
	
	const params: EntityService.Params.ReadWrite<'api::restaurant.restaurant'> = {
	  /* locale should be suggested here */
	  locale: 'fr',
	};	
	```

### Related issue(s)/PR(s)

N/A
